meson.build

```diff
@@ -355,7 +359,7 @@ foreach it : ccs
   have_pthread = it_th.found() and it_machine.system() != 'windows'
   if it_machine.system() == 'sunos'
     # workaround for Solaris until https://github.com/mesonbuild/meson/issues/4328 is fixed
```

Comment:
Remove the comment also then.

Reply:
Can't test unfortunately. Seems my solaris disk is not in the server and I have no physical access at the moment. But from manual review, this looks good and likely we can remove this workaround entirely in another patch independently of this pr.